README.md

```diff
@@ -29,3 +29,126 @@ OpenEBS/project
 ```
 
 > Note: Developer should keep `integration_test` completely decoupled from the rest of the project packages.
+
+## Instantiation
+
+Developer has to instantiate CITF using `NewCITF(configFilePath)` which will initialize the CITF instance with all the required configurations. 
+
+CITF has four components:- 
+- Environment - To Setup or TearDown the platform such as minikube, GKE, AWS etc.
+- K8S - K8S will have Kubernetes ClientSet & Config.
+- Docker - Docker will be used for docker related operations.
+- DebugEnabled - for verbose log.
+
+> Currently CITF supports minikube only.
+
+Developer can pass environment according to their requirements.
+
+By default it will take Minikube as environment.
+
+## Confiuration
+
+To configure the environment of CITF, there are three ways:-
+ - [Environment Variable](#environment-variable)
+ - [Config File](#config-file)
+ - [Default Config](#default-config)
+
+### Environment Variable
+
+At the time of instantiation of CITF, developer can set an environment variable. Environment variable name should be `CITF_CONF_ENVIRONMENT`.
+
+For example:- `export CITF_CONF_ENVIRONMENT = minikube`
+
+### Config File
+If environment variable is not set then developer can pass environment using config file. The file should be in `yaml` format. 
+
+For example:- config.yaml
+
+```
+Environment: minikube
+```
+
+### Default Config
+
+If environment variable and config file are not present, then CITF will take default environment which is minikube.
+
+<details>
+<summary><b>Setting up the Platform</b></summary>
+
+`citf.Environment` will handle operations related to the platforms. 
+
+In order to setup the k8s cluster, developer has to call the `Setup()` method which will bring it up.
+
+Developer can also check the status of the platform using `Status()` method.
+
+Once integration test is completed, developer can delete the setup using `TearDown()` method.
+</details>
+
+
+<details>
+<summary><b>Example</b></summary>
+
+```go
+
+package example
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openebs/CITF"
+)
+
+func TestIntegrationExample(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Integration Test Suite")
+}
+
+// Initializing CITF without config file.
+var CitfInstance, err = citf.NewCITF("")
+
+var _ = BeforeSuite(func() {
+
+	// Setting up the default Platform i.e minikube
+	err = CitfInstance.Environment.Setup()
```

Comment:
Won't it face same problem regarding `K8S` that we talked about?

Reply:
Yes, if `minikube` version is greater than `0.24.1`.

Reply:
for now, can we update this code with the solution you come up with?

Reply:
I think we can check-in this code by specifying minikube version as of now. Other folks are waiting on this example.

Reply:
Done !! Thanks @a4abhishek @satyamz @sagarkrsd